models/user/badge.go

```diff
@@ -60,14 +89,26 @@ func GetBadge(ctx context.Context, slug string) (*Badge, error) {
 
 // UpdateBadge updates a badge based on its slug.
 func UpdateBadge(ctx context.Context, badge *Badge) error {
-	_, err := db.GetEngine(ctx).Where("slug=?", badge.Slug).Update(badge)
+	_, err := db.GetEngine(ctx).Where("slug=?", badge.Slug).Cols("description", "image_url").Update(badge)
 	return err
 }
 
-// DeleteBadge deletes a badge.
+// DeleteBadge deletes a badge and all associated user_badge entries.
 func DeleteBadge(ctx context.Context, badge *Badge) error {
-	_, err := db.GetEngine(ctx).Where("slug=?", badge.Slug).Delete(badge)
-	return err
+	return db.WithTx(ctx, func(ctx context.Context) error {
+		// First delete all user_badge entries for this badge
+		if _, err := db.GetEngine(ctx).
+			Where("badge_id = (SELECT id FROM badge WHERE slug = ?)", badge.Slug).
```

Comment:
Why not use badge.ID directly?

Reply:
It looks like some possible dirty data haven't been removed from the database. Maybe we need a doctor command to fix that.

Reply:
same as above